### PR TITLE
accept _ and . in package names

### PR DESCRIPTION
**Pull request by heavybeard**
closes https://github.com/webpro-nl/knip/issues/789

### CODE REVIEW

File: packages/knip/src/util/modules.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const getPackageNameFromFilePath = (value: string) => {
   return value;
 };
 
-export const isStartsLikePackageName = (specifier: string) => /^@?[a-z0-9]/.test(specifier);
+export const isStartsLikePackageName = (specifier: string) => /^@?[a-z0-9._]/.test(specifier);
```
Comment (webpro) [inline packages/knip/src/util/modules.ts:18]:
```suggestion
export const isStartsLikePackageName = (specifier: string) => /^(@[a-z0-9._]|[a-z0-9])/.test(specifier);
```

Regex could be improved perhaps, but at least we're not allowing non-scoped packages to start with `.` or `_`.

**Review by webpro — COMMENTED** [review]
Thanks @heavybeard! I think your solution allows non-scoped packages to start with `.` or `_` which isn't valid?

File: packages/knip/src/util/modules.ts (hunk 1 above)
Comment (heavybeard) [reply packages/knip/src/util/modules.ts:18]: Thanks to chatGPT, we could improve the regex like this

```suggestion
export const isStartsLikePackageName = (specifier: string) => /^(?:@([._]?[a-z0-9-]{1,213})\/)?[a-z0-9]([a-z0-9-_.]*)(\/[a-z0-9]([a-z0-9-_.]*))*$/gm.test(specifier);
```

Regex tests here https://regex101.com/r/Ukm3Fw/2

Or you could use this package https://www.npmjs.com/package/validate-npm-package-name

Comment (heavybeard) [reply packages/knip/src/util/modules.ts:18]: It's ok for me use your suggested version

**Comment by webpro** [comment]
Thanks a lot @heavybeard! I'll release it soon.

**Comment by webpro** [comment]
:rocket: _This pull request is included in v5.30.6. See [Release 5.30.6](https://github.com/webpro-nl/knip/releases/tag/5.30.6) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._

**Pull request merged**